invalided pseudo-version reported while building operator-sdk

## Bug Report

<!-- 
Note: Make sure to first check the prerequisites that can be found in the main README file!

Thanks for filing an issue! Before hitting the button, please answer these questions.
Fill in as much of the template below as you can. If you leave out information, we can't help you as well.
-->

**What did you do?**
i was following the [Compile and install from master](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/install-operator-sdk.md#compile-and-install-from-master) instructions in the docs and the make failed.

**What did you expect to see?**
i expected the project to build and to see binaries for operator-sdk.

**What did you see instead? Under which circumstances?**
when i reached the command step that instructs to run `make tidy`, i saw this output:

```
$ make tidy
go: github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20191115003340-16619cd27fa5 requires
        github.com/openshift/api@v3.9.1-0.20190924102528-32369d4db2ad+incompatible: invalid pseudo-version: preceding tag (v3.9.0) not found
go: github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20191115003340-16619cd27fa5 requires
        github.com/openshift/api@v3.9.1-0.20190924102528-32369d4db2ad+incompatible: invalid pseudo-version: preceding tag (v3.9.0) not found
make: *** [Makefile:69: tidy] Error 1
```

**Environment**
* operator-sdk version: master @ 7a3da2f203bab0d6cdca5c050a2544295237518b

* go version: `go version go1.13.6 linux/amd64`

<!--- Insert the output of `go version` here -->

* Kubernetes version information: n/a

<!--- Insert the output of `kubectl version` here -->

* Kubernetes cluster kind: n/a

* Are you writing your operator in ansible, helm, or go? n/a

**Possible Solution**
<!--- Only if you have suggestions on a fix for the bug -->

**Additional context**

oh well, i tried deleting my entire `$HOME/go`, and then rebuilding. same result

```
$ make tidy
go: github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20191115003340-16619cd27fa5 requires
        github.com/openshift/api@v3.9.1-0.20190924102528-32369d4db2ad+incompatible: invalid pseudo-version: preceding tag (v3.9.0) not found
go: github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20191115003340-16619cd27fa5 requires
        github.com/openshift/api@v3.9.1-0.20190924102528-32369d4db2ad+incompatible: invalid pseudo-version: preceding tag (v3.9.0) not found
make: *** [Makefile:69: tidy] Error 1
```

--- a/doc/dev/developer_guide.md
+++ b/doc/dev/developer_guide.md
@@ -4,7 +4,7 @@ This document explains how to setup your dev environment.
 
 ## Prerequisites
 - [git][git-tool]
-- [go][go-tool] version v1.12+
+- [go][go-tool] version v1.13+
 
 ## Download Operator SDK
 

--- a/doc/user/install-operator-sdk.md
+++ b/doc/user/install-operator-sdk.md
@@ -86,12 +86,15 @@ $ chmod +x operator-sdk-${RELEASE_VERSION}-x86_64-apple-darwin && sudo mkdir -p 
 - [go][go_tool] version v1.13+.
 
 ```sh
-$ go get -d github.com/operator-framework/operator-sdk # This will download the git repository and not install it
-$ cd $GOPATH/src/github.com/operator-framework/operator-sdk
+$ git clone https://github.com/operator-framework/operator-sdk
+$ cd operator-sdk
 $ git checkout master
 $ make tidy
 $ make install
 ```
+
+**Note:** Ensure that your `GOPROXY` is set with its default value for Go
+versions 1.13+ which is `https://proxy.golang.org,direct`.
 
 [homebrew_tool]:https://brew.sh/
 [git_tool]:https://git-scm.com/downloads

